.github/workflows/semver-checks-pr-label.yml

```diff
@@ -0,0 +1,83 @@
+on:   # yamllint disable-line rule:truthy
+  workflow_run:
+    workflows: [Check semver breaks]
+    types: [completed]
+
+name: Check semver breaks - Label and Comment PR
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  Download:
+    name: Download, Unzip and Add Labels/Comments
+    runs-on: ubuntu-latest
+    # only run if CI passes on the "Check semver breaks" workflow
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    steps:
+      - name: "Download artifact"
+        uses: actions/github-script@v7
+        with:
+          script: |
+            // get all artifacts from the workflow run
+            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: context.payload.workflow_run.id,
+            });
+
+            // find the artifact that starts with 'semver-break'
+            let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name.startsWith('semver-break');
+            })[0];
+
+            // if no artifact found, exit
+            if (!matchArtifact) {
+              console.log('No semver-break artifact found');
+              process.exit(0);
+            }
+
+            // otherwise download the artifact
+            let download = await github.rest.actions.downloadArtifact({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              artifact_id: matchArtifact.id,
+              archive_format: 'zip',
+            });
+
+            // write the artifact to the workspace
+            let fs = require('fs');
+            fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/semver-break.zip`, Buffer.from(download.data));
+      - name: "Unzip artifact"
+        if: ${{ hashFiles('semver-break.zip') != '' }}
+        run: unzip semver-break.zip
+      - name: "Save PR number to env"
+        if: ${{ hashFiles('semver-break') != '' }}
+        run: |
+          PR_NUMBER=$(cat semver-break)
+          echo "PR_NUMBER=$PR_NUMBER" >> $GITHUB_ENV
```

Comment:
Nope, this is vulnerable to arbitrary code execution if someone writes `123\nevil_code_here` into `semver-break`. We need to sanity-check the file to have a single line consisting of only numbers.

Reply:
Good point!

Reply:
Updated to

```
          # check if PR_NUMBER is a number
          if ! [[ $PR_NUMBER =~ ^-?[0-9]+$ ]]; then
              echo "$PR_NUMBER is not a number."
              exit 1
          fi
          echo $PR_NUMBER > ./semver-break
```

Reply:
I am checking in both jobs and also I've tested this change in https://github.com/storopoli/rust-bitcoin/pull/10 and it works.

Reply:
`$PR_NUMBER` needs to be in quotes as not doing so can do something funny with spaces.

Reply:
All quoted thanks for pointing it out.